yb-voyager/src/tgtdb/oracle.go

```diff
@@ -458,8 +530,48 @@ func (tdb *TargetOracleDB) IfRequiredQuoteColumnNames(tableName string, columns
 	return columns, nil
 }
 
+// execute all events sequentially one by one in a single transaction
 func (tdb *TargetOracleDB) ExecuteBatch(migrationUUID uuid.UUID, batch EventBatch) error {
-	// Not implemented
+	// TODO: figure out how to avoid round trips to Oracle DB
+	log.Infof("executing batch of %d events", len(batch.Events))
+
+	err := tdb.WithConn(func(conn *sql.Conn) (bool, error) {
+		tx, err := conn.BeginTx(context.Background(), nil)
+		if err != nil {
+			return false, fmt.Errorf("begin transaction: %w", err)
+		}
+		defer tx.Rollback()
+
+		for i := 0; i < len(batch.Events); i++ {
+			event := batch.Events[i]
+			stmt := event.GetSQLStmt(tdb.tconf.Schema)
+			_, err = tx.Exec(stmt)
+			if err != nil {
+				log.Errorf("error executing stmt: %v", err)
+				return false, fmt.Errorf("error executing stmt %q: %w", stmt, err)
```

Comment:
Include the `vsn` of the `Event` in the message so that in case of an on-call we know where to start the debugging.